### DISCUSSION TITLE
Unpublished Pages in sitetree

### DISCUSSION THREAD

**Discussion by rileywilljohn** [requarks/wiki#4818] [Error / Bug Report]
By Default an _unpublished_ page will be only able to be read (and edited) with the read _and_ write pages permissions added to the group a user belongs to.

However, currently an _unpublished_ page's **title** can still be viewed via the site tree in the sidebar as well as when browsed via tags even when the sidebar is hidden, regardless of the permission set given to a user.

This does not seem like intended behaviour. It looks as though the read permission still grants a user without write permissions the ability to **see** that an unpublished page exist, as well as to click on the link that goes to an error page.

---

### Permissions Via published state:

Page /home/test-hidden is unpublished
![Page Status](https://user-images.githubusercontent.com/28457654/147167743-4ea948ba-59c5-488c-b50b-0655f2be8846.png)

As default the guest group does not have write permissions.
![Screenshot 2021-12-22 at 18-55-16 Admin](https://user-images.githubusercontent.com/28457654/147167874-31f7592f-a8ee-4e30-8e70-1a55de90dfc4.png)

Yet the test-hidden page is still visible in the side bar.
![Screenshot 2021-12-22 at 18-56-24 home](https://user-images.githubusercontent.com/28457654/147167937-cbac14bd-caaa-436a-bc03-3140d0f2e002.png)

Clicking the link obviously leads to the Unauthorized page.

---

### Permission Via Tags:

A different example would be disallowing read permissions via a tag:

Another test page that is **published** with a tag called hidden.
![Screenshot 2021-12-22 at 19-01-13 Test](https://user-images.githubusercontent.com/28457654/147168239-04e2a047-3ba6-46dc-b56d-5ef020959e27.png)

Creating a rule to prevent viewing a page via the tag called hidden.
![Screenshot 2021-12-22 at 19-02-36 Admin](https://user-images.githubusercontent.com/28457654/147168348-65cef680-f135-4336-92e8-7e6229ba34c4.png)

The page is still visible in the sidebar:
![Screenshot 2021-12-22 at 19-06-18 home](https://user-images.githubusercontent.com/28457654/147168574-85a393f1-14ac-4a71-99a7-02bfabd4a15d.png)
As well as in the tag browser:
(test is published, test hidden is unpublished)
![Screenshot 2021-12-22 at 19-09-21 Tags](https://user-images.githubusercontent.com/28457654/147168794-3c77f3d4-00b4-4f02-8465-df6e00cd5279.png)

Both links work and link to the Unauthorized page:

It seems like there needs to be an expanded permission option for reading:unpublished or a user interface change to avoid displaying unpublished pages without write permissions or unreadable pages in the site tree / tag browser.

[Feedback](https://feedback.js.wiki/wiki/p/option-for-unpublished-page-only-visible-for-userowner)

**Comment by FiretronP75** [comment]
I'm newly trying out Wiki.js and immediately noticed this bug. This needs to be fixed before any business can use it in a production environment.

**Comment by FiretronP75** [comment]
I created a page rule for the guest user same as pictured above, except instead of a hidden tag I used a /hidden Path Starts With, and it successfully hid the page in the site tree.

But it should be a default behavior to hide unpublished pages from guests, without having to move or tag them and create a rule for it.

**Comment by FiretronP75** [comment]
Maybe there could be a new permission called "Can View Unpublished in Navigation"
